operatorcourier/format.py

```diff
@@ -0,0 +1,86 @@
+import yaml
+from operatorcourier.build import BuildCmd
+
+class _literal(str): pass
+
+def _literal_presenter(dumper, data):
+    return dumper.represent_scalar('tag:yaml.org,2002:str', data, style='|')
+
+def _get_empty_formatted_bundle():
+    return dict(
+        data = dict(
+            customResourceDefinitions = '',
+            clusterServiceVersions = '',
+            packages = '',
+        )
+    )
+
+def format_bundle(bundle):
+    """
+    Converts a bundle object into a push-ready bundle by changing list values of 'customResourceDefinitions', 'clusterServiceVersions', and 'packages' into stringified yaml literals.
+    This format is required by the Marketplace backend.
+    
+    :param bundle: A bundle object
+    """
+
+    formattedBundle = _get_empty_formatted_bundle()
+
+    yaml.add_representer(_literal, _literal_presenter)
+
+    if 'data' in bundle:
+        # Format data fields as string literals to match backend expected format
+        if 'customResourceDefinitions' in bundle['data']:
+            if len(bundle['data']['customResourceDefinitions']) > 0:
+                formattedBundle['data']['customResourceDefinitions'] = _literal(yaml.dump(bundle['data']['customResourceDefinitions'], default_flow_style=False))
+
+        if 'clusterServiceVersions' in bundle['data']:
+            # Format description and alm-examples
+            clusterServiceVersions = []
+            for csv in bundle['data']['clusterServiceVersions']:
+                if 'metadata' in csv:
+                    if 'annotations' in csv['metadata']:
+                        if 'alm-examples' in csv['metadata']['annotations']:
```

Comment:
Formatting `alm-examples` and `description` isn't required for proper parsing by the marketplace operator, but I added it because (1) it's more readable in the resulting bundle and (2) I thought it would help with consistent format for the main two multiline string fields as part of the current test suite. Since it's not required I have no objections removing the check for these two fields, as long as tests can still pass.
I think it's worthwhile to create such a method that checks a path exists in a dict, would be a nice util function (if it doesn't exist in the standard lib).

Reply:
Okay - I'll try and get around to adding the method in the future.